File: src/main/java/ordilov/lissn/member/application/service/GetMemberService.java
package ordilov.lissn.member.application.service;

import java.util.Optional;
import lombok.RequiredArgsConstructor;
import ordilov.lissn.member.application.port.in.GetMemberQuery;
import ordilov.lissn.member.application.port.out.MemberReader;
import ordilov.lissn.member.domain.MemberInfo.GetMemberInfo;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public class GetMemberService implements GetMemberQuery {

  private final MemberReader memberReader;

  @Override
  public GetMemberInfo getMember(Long memberId) {
    return new GetMemberInfo(memberReader.getMemberBy(memberId));
  }

  @Override
  public Optional<GetMemberInfo> getMemberByEmail(String email) {
    return memberReader.getMemberByEmail(email).map(GetMemberInfo::new);
  }
}

File: src/main/java/ordilov/lissn/member/application/service/GetMemberService.kt
package ordilov.lissn.member.application.service;

import MemberReader
import ordilov.lissn.member.application.port.`in`.GetMemberQuery
import ordilov.lissn.member.domain.GetMemberInfo
import org.springframework.stereotype.Service

@Service
class GetMemberService(
    val memberReader: MemberReader
) : GetMemberQuery {

    override fun getMember(memberId: Long): GetMemberInfo {
        val member = memberReader.getMemberBy(memberId)
        return GetMemberInfo(
            member.id!!,
            member.name,
            member.email,
            member.picture,
            member.provider
        )
    }

    override fun getMemberByEmail(email: String): GetMemberInfo {
        val member = memberReader.getMemberByEmail(email)
        return GetMemberInfo(
            member.id!!,
            member.name,
            member.email,
            member.picture,
            member.provider
        )
    }
}